### PR TITLE
feat: ✨ adds bun 🧅

### PR DESCRIPTION
**Pull request by fpigeonjr**
## What changed

feat: ✨ adds bun 🧅

## Issue

#1111 

## How to test

_Write out steps for how someone could test this PR against the acceptance criteria_

## Screenshots

_If relevant, e.g. for a front-end feature_

## Definition of Done Checklist
- [-] OESA: Code refactored for clarity
- [-] OESA: Dependency rules followed
- [X] Automated unit tests updated and passed
- [X] Automated integration tests updated and passed
- [X] Automated quality tests updated and passed
- [X] Automated load tests updated and passed
- [X] Automated a11y tests updated and passed
- [X] Automated security tests updated and passed
- [X] 90%+ Code coverage achieved
- [-] Form validations updated

## Links

[Bun docs](https://bun.sh)

### CODE REVIEW

**Comment by fpigeonjr** [comment]
have some lingering issues with `webpack-dev-server`
SCR-20231103-jpaa

**Comment by jonnalley** [comment]
👏 👏 🍹

**Comment by fpigeonjr** [comment]
😞
SCR-20231106-mqgi

**Comment by fpigeonjr** [comment]
> 😞 SCR-20231106-mqgi

this has been resolved

**Comment by fpigeonjr** [comment]
> have some lingering issues with `webpack-dev-server`:
> 
> SCR-20231103-jpaa

well this is something
image

**Comment by fpigeonjr** [comment]
I am seeing this warning in the terminal but everything seems to be working fine 🤷 

```bash
ops-frontend     |       at /home/app/node_modules/webpack-dev-server/lib/servers/WebsocketServer.js:70:12
ops-frontend     |       at forEach (:1:20)
ops-frontend     |       at /home/app/node_modules/webpack-dev-server/lib/servers/WebsocketServer.js:64:6
ops-frontend     | 65 |         /**
ops-frontend     | 66 |          * @param {ClientConnection} client
ops-frontend     | 67 |          */
ops-frontend     | 68 |         (client) => {
ops-frontend     | 69 |           if (client.isAlive === false) {
ops-frontend     | 70 |             client.terminate();
ops-frontend     |                 ^
ops-frontend     | TypeError: client.terminate is not a function. (In 'client.terminate()', 'client.terminate' is undefined)
ops-frontend     |       at /home/app/node_modules/webpack-dev-server/lib/servers/WebsocketServer.js:70:12
ops-fro
```

**Review by johndeange — APPROVED** [review]

**Review by stevtek — APPROVED** [review]

File: .github/actions/setup-javascript/action.yml [hunk 1]
```diff
@@ -3,13 +3,12 @@ description: Installs Node.js and dependencies defined in package.json
 runs:
   using: composite
   steps:
-    - name: Set up node
-      uses: actions/setup-node@8f152de45cc393bb48ce5d89d36b731f54556e65 # tag: v3.5.1
+    - name: Set up Bun
+      uses: oven-sh/setup-bun@v1
```
Comment (tdonaworth) [inline .github/actions/setup-javascript/action.yml:7]: can we digest pin this please?

File: frontend/Dockerfile [hunk 2]
```diff
@@ -1,11 +1,13 @@
-FROM node:20-slim
+FROM oven/bun:latest
```
Comment (tdonaworth) [inline frontend/Dockerfile:1]: Have we tried this with 'oven/bun:slim' ? I haven't done a comparison 1:1 yet with node:20-slim vs bun:slim - but optimally I'd prefer to run the leanest base image we can (fewer dependencies, etc).

File: .github/actions/setup-javascript/action.yml [hunk 3]
```diff
@@ -1,15 +1,14 @@
...
-    - name: Set up node
-      uses: actions/setup-node@ee36e8b5c0fdd6014a0398aed18ce9876360bd63 # tag: v3.5.1
+    - name: Set up Bun
+      uses: oven-sh/setup-bun@a1800f471a0bc25cddac36bb13e6f436ddf341d7 # v1
       with:
-        node-version-file: ./frontend/.nvmrc
-        cache: 'yarn'
-        cache-dependency-path: '**/yarn.lock'
-    - name: Install yarn dependencies
+        bun-version: latest
```
Comment (tdonaworth) [inline .github/actions/setup-javascript/action.yml:9]: We want to run the same version we're using in our app here - simply for consistency; so in this case it'd be `489fa7b8b7a2f1dd58968bdf5b360fa144ac1d30bdcfdd98cdda33e33e6fbf22`

**Pull request merged**